plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1130,6 +1135,14 @@ public ConnectorInsertTableHandle beginInsert(ConnectorSession session, Connecto
         return newWritableTableHandle(table.getSchemaTableName(), icebergTable, retryMode);
     }
 
+    private boolean hasNestedNamespaces(ConnectorSession session, String parentNamespace)
+    {
+        return catalog.listNamespaces(session)
+                .stream()
+                .filter(namespace -> !namespace.equals(parentNamespace))
+                .anyMatch(namespace -> namespace.startsWith(parentNamespace));
```

Comment:
This logic looks wrong. The namespace without separator shouldn't be treated as nested namespaces. e.g. `tpch` vs `tpch1`

Reply:
Thanks, fixed that